### ISSUE TITLE
Most users not read from active directory with known working ldapfilter

### ISSUE THREAD

**Issue by SCRICrtoole** [adobe-apiplatform/user-sync.py#467]
user-sync 2.4 running on RHEL 7.6 connecting to Active Directory

connector-ldap.yml configured to use default search string  
all_users_filter: "(&(objectClass=user)(objectCategory=person))

base_dn: configured for the root of the domain

The configuration shown above should return all users in the domain.  It only returns 1520 users.  The exact same settings in an ldap browser using the same credentials returns 6657 users.  

I have attached a screenshot of both the user-sync output and the ldap search.

![user-sync](https://user-images.githubusercontent.com/40764804/54361890-5cc76980-463e-11e9-82c5-45e76afad7a1.png)

![LDP](https://user-images.githubusercontent.com/40764804/54361069-9d25e800-463c-11e9-9e2e-28a515ea0b58.png)

**Comment by vossen-adobe** [comment]
Can you please provide additional info such as:

Command line options - are you using users mapped?  Do the groups contain all of the users?
AD structure - are you trying to query multiple domains?

**Comment by SCRICrtoole** [comment]
Thanks for the quick follow-up. Currently, I am only running in test mode as I wanted to get everything working properly before creating the users.
I have tried with no switches other than --test-mode
./user-sync --test-mode
I have also tried with the following
./user-sync --users mapped --process-groups --test-mode

I am only connecting to a single domain although we do have multiple.  All the users referenced in the examples are in a single domain.

I looked at my group mappings and one of the groups that users were in was not mapped.  After mapping that group the users were all read. 

Looks like this has resolved it.  All the documentation seems to explain how to map your groups etc, but doesnt seem to specify that a user must be in a mapped group to be read.

Thanks for your help!! Closing
